### PR TITLE
feat: supporting jinja templating in saved metrics 

### LINKED ISSUE #6983: Jinja templating is ignored in Metrics definitions
Make sure these boxes are checked before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X] I have reproduced the issue with at least the latest released version of superset.
- [X] I have checked the issue tracker for the same issue and I haven't found one similar.

### Superset version
0.29.0rc7 (latest on pypi)

### Expected results
Not sure if this is intentional, but it seems like Jinja templating used when defining metrics using custom SQL should be parsed and rendered similarly to how it works when defining filters.

### Actual results
Jinja templating in metrics is effectively ignored and sent to the query engine as is.

### Steps to reproduce
Add a metric to a chart that includes Jinja templating. For example, define a metric as `max('{{ from_dttm }}')`. The rendered SQL shows that the templating is ignored.

### PR DESCRIPTION
**Pull request by guydou**
### SUMMARY
Adding the ability to use jinja templating also in saved metrics. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
image

### TESTING INSTRUCTIONS
1. A saved metric that uses a jinja expression
2. Go to Chart and use that saved metric
3. Look in the view query, and make sure the jinja expression was evaluated correctly

### ADDITIONAL INFORMATION

- [X] Has associated issue:
   - Fixes: #6983
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/connectors/sqla/models.py [hunk 1]
```diff
@@ -404,9 +404,14 @@ class SqlMetric(Model, BaseMetric):
...
 
+    def get_template_processor(self, **kwargs: Any) -> BaseTemplateProcessor:
+        return self.table.get_template_processor()
+
     def get_sqla_col(self, label: Optional[str] = None) -> Column:
         label = label or self.metric_name
-        sqla_col: ColumnClause = literal_column(self.expression)
+        sqla_col: ColumnClause = literal_column(
+            self.get_template_processor().process_template(self.expression)
+        )
```
Comment (zhaoyongjie) [inline superset/connectors/sqla/models.py:414]: It is not recommended to add the new instance method(get_template_processor) here.

```suggestion
    def get_sqla_col(self, label: Optional[str] = None) -> Column:
        label = label or self.metric_name

        tp = self.table.get_template_processor()
        sqla_col = literal_column(tp.process_template(self.expression))
        return self.table.make_sqla_column_compatible(sqla_col, label)
```

**Comment by zhaoyongjie** [comment]
closes: https://github.com/apache/superset/issues/6983

File: superset/connectors/sqla/models.py (hunk 1 above)
Comment (guydou) [reply superset/connectors/sqla/models.py]: Got it, 

Fixed the code, 

Why it is not recommended to have new instance method? I am asking only yo expand my knowledge

Thanks

Comment (zhaoyongjie) [reply superset/connectors/sqla/models.py]: Hi Guyt. The main reason is no more `self.get_template_processor` method call currently, so no need to extend this class. If there is a need for this in the future, we can easily extend it.

**Review by zhaoyongjie — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**